src/webui/api/torrentscontroller.cpp

```diff
@@ -250,6 +251,64 @@ namespace
             idList << BitTorrent::TorrentID::fromString(hash);
         return idList;
     }
+
+    QJsonObject serializeTorrentMetadata(const BitTorrent::TorrentInfo &metadata, const QVector<BitTorrent::TrackerEntry> &trackers)
+    {
+        QJsonArray trackersArr;
+        for (const auto &tracker : trackers)
+        {
+            trackersArr << QJsonObject
+            {
+                {KEY_TRACKER_URL, tracker.url},
+                {KEY_TRACKER_TIER, tracker.tier}
+            };
+        }
+
+        QJsonArray files;
+        for (const auto &filePath : metadata.filePaths())
```

Comment:
The below code isn't optimal. It's better to rewrite the loop as the following:
```suggestion
        for (int fileIndex = 0; fileIndex < metadata.file count(); ++fileIndex)
```

Reply:
This will allow changing `TorrentInfo::fileIndex()` back to private.